src/api/usePointsBalance.ts

```diff
@@ -1,26 +1,37 @@
-import { useQuery } from "@tanstack/react-query";
+import { useCallback, useEffect, useState } from "react";
 import { useLoginStateController } from "../Login";
 
 export const POINTS_BALANCE_KEY = "POINTS_BALANCE";
 
 export function usePointsBalance() {
   const { userId } = useLoginStateController();
+  const [points, setPoints] = useState<number>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>();
 
-  return useQuery({
-    queryKey: [POINTS_BALANCE_KEY, userId],
-    queryFn: async () => {
+  const getData = useCallback(async (id: string) => {
+    setError(undefined);
+    setIsLoading(true);
+
+    try {
       const response = await fetch(
-        `http://localhost:8080/points/balance/${userId}`
+        `http://localhost:8080/points/balance/${id}`
       );
+      const data: { balance: number } = await response.json();
 
-      if (!response.ok) {
```

Comment:
I wouldn't remove the ability for the backend to send back explicit error messages and have them handled here.

I suggest this alternative implementation:

```ts
    setError(undefined);
    setIsLoading(true);

    try {
      const response = await fetch(
        `http://localhost:8080/points/balance/${id}`
      );

      if (!response.ok) { 
        throw new Error(await response.json());
      }
      
      const data: { balance: number } = await response.json();
      setPoints(data.balance);
    } catch (e: any) {
      setError(e.message);
    } finally {
      setIsLoading(false);
    }
```